pkg/pillar/cmd/volumemgr/handleverifier.go

```diff
@@ -58,11 +69,150 @@ func MaybeAddVerifyImageConfig(ctx *volumemgrContext,
 			}
 		}
 	}
-	AddOrRefcountVerifyConfig(ctx, &status)
+
+	verifyConfig := lookupVerifyImageConfig(ctx, status.ObjType, status.BlobSha256)
+	if verifyConfig != nil {
+		verifyConfig.RefCount++
+		log.Infof("MaybeAddVerifyImageConfig: refcnt to %d for %s",
+			verifyConfig.RefCount, status.BlobSha256)
+		publishVerifyImageConfig(ctx, status.ObjType, verifyConfig)
+	} else {
+		log.Infof("MaybeAddVerifyImageConfig: add for %s, IsContainer: %t",
+			status.BlobSha256, status.DownloadOrigin.IsContainer)
+		n := types.VerifyImageConfig{
+			ImageID: status.VolumeID,
+			VerifyConfig: types.VerifyConfig{
+				Name:             status.DisplayName,
+				ImageSha256:      status.BlobSha256,
+				CertificateChain: status.DownloadOrigin.CertificateChain,
+				ImageSignature:   status.DownloadOrigin.ImageSignature,
+				SignatureKey:     status.DownloadOrigin.SignatureKey,
+			},
+			IsContainer: status.DownloadOrigin.IsContainer,
+			RefCount:    1,
+		}
+		if status.State == types.VERIFYING {
+			_, err := markObjectAsVerifying(&n, status.ObjType)
+			if err != nil {
+				return false, types.ErrorAndTime{
+					Error:     err.Error(),
+					ErrorTime: time.Now(),
+				}
+			}
+		}
+		publishVerifyImageConfig(ctx, status.ObjType, &n)
+		log.Debugf("MaybeAddVerifyImageConfig - config: %+v", n)
+	}
 	log.Infof("MaybeAddVerifyImageConfig done for %s", status.BlobSha256)
 	return true, types.ErrorAndTime{}
 }
 
+// Returns ok, size of object
+func markObjectAsVerifying(config *types.VerifyImageConfig, objType string) (int64, error) {
+
+	pendingDirname, verifierDirname, _ := ImageDownloadDirNames(objType, config.ImageID.String(), config.ImageSha256)
+	pendingFilename, verifierFilename, _ := ImageDownloadFilenames(objType, config.ImageID.String(), config.ImageSha256, config.Name)
+
+	// Move to verifier directory which is RO
+	// XXX should have dom0 do this and/or have RO mounts
+	log.Infof("markObjectAsVerifying: Move from %s to %s", pendingFilename, verifierFilename)
+
+	info, err := os.Stat(pendingFilename)
+	if err != nil {
+		e := fmt.Errorf("markObjectAsVerifying: Unable to find location: %s. %s", pendingFilename, err)
+		// XXX hits sometimes; attempting to verify before download
+		// is complete?
+		log.Errorf(e.Error())
+		return 0, e
+	}
+
+	if _, err := os.Stat(verifierFilename); err == nil {
+		log.Warnf("markObjectAsVerifying: %s : file exists. Deleting it", verifierFilename)
+		if err := os.RemoveAll(verifierFilename); err != nil {
```

Comment:
I am not sure this is correct. Why should we have to remove the directory? The work you are doing here converts verifier from working on _directories_ to working on _files_. Nothing prevents us from having multiple such files in a single directory. And actually, that is precisely what we plan on doing with container image downloads by parts.

Reply:
@adarsh-zededa convinced me. We will leave this for now, add capability to handle multiple files in a single image directory in a future one.

Reply:
If you have volumemgr specify Target or TargetBaseName, then you only care about that file existing and not about the directory.